### PR TITLE
Implement login form and router.

### PR DESCRIPTION
**Pull request by Oleh-Stefanovych**
* Implementation of form for user log in. It has fields login and password. 
* Implementation of router. It routes to the specified page using a hash. Supports login page, registration page and 404 page.

### CODE REVIEW

**Comment by Oleh-Stefanovych** [comment]
@dkalinin, PTAL.

**Comment by Oleh-Stefanovych** [comment]
ESLint result:
image

File: app/component/hyperlink/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+   * Note: at the time of applying the configuration by the method
+   * {@link Component#applyConfiguration} or constructor, rendering is disabled. To avoid
+   * unnecessary rendering operations when overriding this method, the rendering enabled should
+   * be checked:
+   * <code>
+   *    if (this._isRenderingEnabled) {
+   *       //render.
+   *    }
+   * </code>
+   * </p>
```
Comment (dkalinin) [inline app/component/hyperlink/index.js]: Check this: https://jsdoc.app/tags-inheritdoc.html

File: app/component/page-404/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  /**
+   * Performs initial rendering of the component and creates subcomponents. May be overridden,
+   * but this will require calling the parent method: <code> super._initialRender(); </code>.
+   *
+   * @private
+   */
+  _initialRender() {
+    new Component(this.element, 'span').element.textContent = 'Page Not Found!';
```
Comment (dkalinin) [inline app/component/page-404/index.js]: Please add link to the application's root.

**Pull request merged**